src/UniswapV4DeployerCompetition.sol

```diff
@@ -0,0 +1,87 @@
+// SPADIX-License-Identifier: UNLICENSED
+pragma solidity 0.8.26;
+
+import {Create2} from "@openzeppelin/contracts/utils/Create2.sol";
+import {Owned} from "solmate/src/auth/Owned.sol";
+import {VanityAddressLib} from "./libraries/VanityAddressLib.sol";
+import {IUniswapV4DeployerCompetition} from "./interfaces/IUniswapV4DeployerCompetition.sol";
+
+/// @title UniswapV4DeployerCompetition
+/// @notice A contract to crowdsource a salt for the best Uniswap V4 address
+contract UniswapV4DeployerCompetition is IUniswapV4DeployerCompetition {
+    using VanityAddressLib for address;
+
+    /// @dev The salt for the best address found so far
+    bytes32 public bestAddressSalt;
+    /// @dev The best address found so far
+    address public bestAddress;
```

Comment:
GAS

Not necessary to store since it can always be derived from the salt, which is stored.

Reply:
same for `bestAddressSubmitter` if one of two conditions met:

- the value is not needed on chain in another contract
- `saltSubAddress` not permitted to be anything other than `msg.sender`

Reply:
> GAS
> 
> Not necessary to store since it can always be derived from the salt, which is stored.

I agree with this, it could be converted to a public function:
```solidity
function bestAddress() external view returns (address) {
  return Create2.computeAddress(bestAddressSalt, initCodeHash);
}
```

Reply:
good call - fixed here! https://github.com/Uniswap/v4-periphery/pull/117/commits/fbd159640ead8792b33bc309cd5071235c41f949

Reply:
good call - fixed here! https://github.com/Uniswap/v4-periphery/pull/117/commits/fbd159640ead8792b33bc309cd5071235c41f949